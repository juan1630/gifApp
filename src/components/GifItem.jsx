export const GifItem = ({title, url})=> {
    return(
        <div className="card-gifs" >
            <img src={url} alt={title} />
            <p> { title } </p>
        </div>
    )
}