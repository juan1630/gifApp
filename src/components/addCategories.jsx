import { useState } from "react"

export const AddCategory = ({ onAddCategory  }) => {
    
    const [inputValue, setInputValue ] = useState("");

    const onChangeInputValue = (event) => {
        // console.log(event);
        setInputValue(event.target.value );
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if( inputValue.trim().length <= 2) return; 
        // setCategories((categories) => [inputValue, ...categories] );
        onAddCategory(inputValue.trim());
        //emitimos el newCategory
        setInputValue('');
    }

    return ( 
        <>
            <form  onSubmit={ (event)=> onSubmit(event) } > 
                <h2> Add category </h2> 
                <input 
                    type="text" 
                    placeholder="Buscar gifs"
                    value={inputValue}
                    onChange={ (event)=> onChangeInputValue(event)}
                    />
            </form>
        </>
    )
}