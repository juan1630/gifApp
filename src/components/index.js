export * from './GifItem';
export * from './addCategories';