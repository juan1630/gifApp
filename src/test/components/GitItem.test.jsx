import {render, screen} from '@testing-library/react'
import { GifItem } from '../../components/GifItem';

describe('Pruebas en el componente GifItem' , () => {

    const title="Attack on titan";
    const url = 'https://www.attack.com/';

    test('Debe de hacer match con el  snapshot', ()=> {

        

        const {container} = render(<GifItem title={title} url={url} />);
        

        expect( container ).toMatchSnapshot();
        //compramos que el componente este igual a como esta la foto del componente 
    });

     test('Debe de realizar una prueba y devolver una imagen con el url y el alt indicado ', () => {

        render(<GifItem title={title} url={url} />);
        // srive para poder ver en la cosnola el componente al que se le hacen las pruebas
        // screen.debug();
        // console.log( screen.getByRole('img').src );

        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);

        const  { src, alt  } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    
    });


    test('debe de mostrar el titulo en el componente', ()=> {
        render( <GifItem  title={title} url={url} /> );
        expect( screen.getByText ).toBeTruthy();
    });
});