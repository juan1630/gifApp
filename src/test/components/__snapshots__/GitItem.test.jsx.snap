// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe de hacer match con el  snapshot 1`] = `
<div>
  <div
    class="card-gifs"
  >
    <img
      alt="Attack on titan"
      src="https://www.attack.com/"
    />
    <p>
       
      Attack on titan
       
    </p>
  </div>
</div>
`;
